packages/execution-environments/src/common/keyring.ts

```diff
@@ -21,29 +27,33 @@ export function wrapKeyring(
     throw new Error('Keyring not exported');
   }
 
-  const keyringHandler = ({
-    request,
-  }: {
-    request: JsonRpcRequest<unknown[]>;
-  }) => {
+  const keyringHandler = ({ request }: { request: JsonRpcRequest<Json[]> }) => {
     const { method, params } = request;
     if (!(method in keyring)) {
       throw new Error(`Keyring does not expose ${method}`);
     }
-    let args = params ?? [];
-    // @ts-expect-error TODO: Figure out how to type this better
-    const func = keyring[method].bind(keyring);
+    let args = (params ?? []) as unknown[];
```

Comment:
Would this work? This could potentially remove the type cast further on.
```suggestion
    let args: Json[] = params ?? [];
```

Reply:
No, because on line 54 assign a function callback